components/map/basic/src/leaflet/marker-manager.js

```diff
@@ -0,0 +1,179 @@
+import L from 'leaflet'
+class MarkerManager {
+  constructor (mapId) {
+    this.setMapDOMInstance(mapId)
+    this.setMarkerDefaults()
+    this.HEART_ICON = '&#9829;'
+  }
+
+  setMapDOMInstance (id) {
+    this.mapDOM = document.getElementById(id)
+  }
+
+  dispatchCustomEvent ({ eventName, detail }) {
+    let event
+
+    if (this.mapDOM.CustomEvent && typeof this.mapDOM.CustomEvent === 'function') {
+      event = new this.mapDOM.CustomEvent(eventName, { detail })
+    } else {
+      event = document.createEvent('CustomEvent')
+      event.initCustomEvent(eventName, true, true, detail)
+    }
+
+    this.mapDOM.dispatchEvent(event)
+  }
+
+  createMarker (item) {
+    const events = [
+      { eventName: 'click', eventHandler: (e) => this.isPoiClicked(e) },
+      { eventName: 'mouseover', eventHandler: (e) => this.onMouseOver(e) },
+      { eventName: 'mouseout', eventHandler: (e) => this.onMouseOut(e) }
+    ]
+    const {latitude, longitude, isSelected, markerType, propertyInfo} = item
+    const marker = L.marker([latitude, longitude], {icon: this.getIconFor({item})})
+    marker.propertyInfo = propertyInfo
```

Comment:
This part is completely coupled to FC contract for POI's this should be changed on the future.

Reply:
Not a fan of having this in sui-components... I thing it should not be allowed.
The follwoing lines are event more coupled with IsFavorite, isPromotion, etc.